include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -0,0 +1,1523 @@
+ */
+template <typename Lyt, typename Ntk>
+using node_dict_type = mockturtle::node_map<mockturtle::signal<Lyt>, Ntk>;
+/**
+ * This struct holds two boolean values indicating the allowed positions for PIs.
+ */
+struct pi_locations
+{
+    /**
+     * Flag indicating if primary inputs (PIs) can be placed at the top.
+     */
+    bool top;
+    /**
+     * Flag indicating if primary inputs (PIs) can be placed at the left.
+     */
+    bool left;
+};
+/**
+ * A structure representing a search space graph.
+ *
+ * This struct encapsulates all the necessary data for managing a search space graph
+ * during the graph-oriented layout design process. It holds the current vertex,
+ * network, nodes to be placed, and other relevant information.
+ *
+ * @tparam Lyt The layout type.
+ * @tparam Ntk The network type.
+ */
+template <typename Lyt>
+struct search_space_graph
+{
+    /**
+     * The current vertex in the search space graph.
+     */
+    coord_vec_type<Lyt> current_vertex{};
+    /**
+     * The network associated with this search space graph.
+     */
+    tec_nt network{};
+    /**
+     * Topological list of nodes to be placed in the layout.
+     */
+    std::vector<mockturtle::node<tec_nt>> nodes_to_place{};
+    /**
+     * Flags indicating if primary inputs (PIs) can be placed at the top or left.
+     */
+    pi_locations pi_locs = {false, false};
+    /**
+     * Flag indicating if this graph's frontier is active.
+     */
+    bool frontier_flag = false;
+    /**
+     * The cost so far for reaching each vertex in the layout.
+     */
+    std::unordered_map<coord_vec_type<Lyt>, double, detail::nested_vector_hash<Lyt>> cost_so_far{};
+    /**
+     * The maximum number of positions to be considered for expansions.
+     */
+    uint64_t num_expansions = 4;
+    /**
+     * Priority queue containing vertices of the search space graph.
+     */
+    detail::priority_queue<Lyt> frontier{};
+};
+/**
+ * This struct holds a single `timeout` value which is determined based on the
+ * provided optional timeout value and a boolean indicating if high effort is required.
+ */
+struct timeout_settings
+{
+    /**
+     * The timeout value in milliseconds.
+     */
+    uint64_t timeout;
+    /**
+     * This constructor initializes the `timeout` member. If `timeout_opt` has a value,
+     * it uses that value for the timeout. If `timeout_opt` does not have a value,
+     * it sets the timeout to 1000000 milliseconds (1000 seconds) if `high_effort` is `true`,
+     * or to 10000 milliseconds (10 seconds) if `high_effort` is false.
+     *
+     * @param timeout_opt An optional unsigned integer value representing the desired timeout in milliseconds.
+     * @param high_effort A boolean indicating whether a high effort (longer timeout) is desired.
+     */
+    timeout_settings(const std::optional<uint64_t>& timeout_opt, const bool high_effort)
+    {
+        timeout = timeout_opt.value_or(high_effort ? 1000000u : 1000u);
```

Comment:
After some testing, I found 1000 ms as the default runtime value quite insufficient. Even some examples from the TOY folder cannot be generated within 1 s. Ideally, GOLD always produces a result unless a timeout is explicitly given. Where `high_effort_mode` might explore more paths and/or handle more queues additionally. I think "low-effort mode" should at least return something (again, unless a low timeout is explicitly passed).